### ISSUE TITLE
Minimal, integrated UI

### ISSUE THREAD

**Issue by arose** [arose/ngl#159]
A simple UI integrated into the NGL core, limited to the canvas. Possible a context menu?

- picked atom info test
- download screenshot

what else?

**Comment by hainm** [comment]
cool. I am playing with `ipywidgets` stuff and am able to use jquery to select `ipywidgets` Widget and add to `NGL` container.

![reorg](https://cloud.githubusercontent.com/assets/4451957/16568525/5abad876-41f9-11e6-8d3d-26c9bad6d3bc.png)

just for fun because I need to get the tab ID from google inspect.

**Comment by hainm** [comment]
- [ ] center
- [ ] representation

**Comment by arose** [comment]
@hainm what do you mean with "representation"

**Comment by hainm** [comment]
>  "representation"

display some sliders so we can change it?

![reorg](https://cloud.githubusercontent.com/assets/4451957/16568667/f51edb28-41fa-11e6-9069-eb265946b1a6.png)

**Comment by fredludlow** [comment]
+1 For the integrated UI - to be honest the existing right-hand panel in the web app is already great - if there was an easy way to inject that into a div (that I could then collapse/expand) that would be fantastic!

**Comment by hainm** [comment]
If possible, please keep it compatible with `nglview` too.

**Comment by arose** [comment]
> to be honest the existing right-hand panel in the web app is already great - if there was an easy way to inject that into a div (that I could then collapse/expand) that would be fantastic!

In addition, otherwise I fear it is too complicated. It exposes almost all of the API.

**Comment by arose** [comment]
> If possible, please keep it compatible with nglview too.

As it would be inside of the canvas this shouldn't be a problem.

**Comment by hainm** [comment]
can be an issue if you allow user to add new structure as trajectory from the canvas

I agree with you that UI in canvas should be very simple. For full UI, there is already the right panel.
But there should be an option to turn on/off the UI.

**Comment by arose** [comment]
> can be an issue if you allow user to add new structure as trajectory from the canvas

I agree, though this could be handled as the is a signal that a trajectory has been added (nglview could subscribe to that).

> But there should be an option to turn on/off the UI.

definitely!

**Comment by hainm** [comment]
The color of the menu control looks nice (to me): https://stemkoski.github.io/Three.js/GUI.html

**Comment by hainm** [comment]
menu example for above link with `xtal.js`

http://natechols.github.io/xtal.js/map_viewer.html

**Comment by hainm** [comment]
When you are done this this for `NGL`, I think I will try to add Jupyter notebook cell to the widget screen too (for auto-completion).

**Comment by hainm** [comment]
this menu looks nice

http://moleculeviewer.bionano.autodesk.com/#

![jt2](https://cloud.githubusercontent.com/assets/4451957/17268318/d41988f2-55f4-11e6-96f4-586813ff1519.png)

**Comment by hainm** [comment]
At some points, I would like to reuse `ngl` UI for `nglview`, like

![share](https://cloud.githubusercontent.com/assets/4451957/17344925/e3f14f32-58d1-11e6-9833-5137d13e341e.png)

(related to https://github.com/arose/ngl/issues/171)
